src/scanner.cc

```diff
@@ -56,7 +56,10 @@ bool is_inline_whitespace(int32_t c) {
 }
 
 bool is_newline(int32_t c) {
-  return c == '\n';
+  // Note: this implies \r\n is treated as two line breaks,
+  // but in our case it's fine, since multiple line breaks
+  // make no difference
+  return c == '\n' || c == '\r';
```

Comment:
The integration tests found a new edge case when parsing [axon example](https://github.com/elixir-nx/axon/blob/main/examples/vision/horses_or_humans.exs) :)